### DISCUSSION TITLE
[solved] HMS-1600 / HMS-1000 openDTU v24.4.12 | Nothing received, resend whole request

### DISCUSSION THREAD

**Discussion by Landorin-GH** [tbnobody/OpenDTU#1930] [General]
I got the same issue that was reported [here ](https://github.com/tbnobody/OpenDTU/discussions/1929) but with HMS inverters since a few days.

I got an opendtu unit (bought this [model](https://www.derdtushop.de/shop/OpenDTU-Open-DTU-fur-Hoymiles-HMS-300-2000-und-HMT-1800-2250-Display-&-Gehause-p603678061)) since a few months and it went fine 24/7, Got 1 HMS-1600 (4 trackers) and an HMS-1000 here. Then a few days ago I got the exact same issue. I see in the log that opendtu transmits data, but does not receive anything. I did not touch the opendtu anymore when I set it up as I only use it to have data transmitted to Home Assistant.

Log (I changed the inverter serial): 
_10:14:12.945 > Network connected
10:14:12.945 > Websocket: [/livedata][1] connect
10:14:12.945 > Websocket: [/livedata][2] connect
10:14:12.945 > Fetch inverter: 1234567890
10:15:18.042 > Websocket: [/livedata][1] disconnect
10:15:18.042 > TX RealTimeRunData 865.00 MHz --> 15 94 40 13 03 80 11 91 44 80 0B 00 66 23 79 55 00 00 00 00 00 00 00 00 91 9C 7A 
10:16:23.155 > Websocket: [/livedata][3] connect
10:16:23.155 > RX Period End
10:16:23.155 > All missing
10:16:23.155 > Nothing received, resend whole request
10:16:23.155 > TX RealTimeRunData 865.00 MHz --> 15 94 40 13 03 80 11 91 44 80 0B 00 66 23 79 55 00 00 00 00 00 00 00 00 91 9C 7A 
10:17:28.267 > Websocket: [/livedata][2] disconnect
10:17:28.267 > RX Period End
10:17:28.267 > All missing
10:17:28.267 > Nothing received, resend whole request
10:17:28.267 > TX RealTimeRunData 865.00 MHz --> 15 94 40 13 03 80 11 91 44 80 0B 00 66 23 79 55 00 00 00 00 00 00 00 00 91 9C 7A 
10:18:33.379 > RX Period End
10:18:33.379 > All missing
10:18:33.379 > Nothing received, resend whole request
10:18:33.379 > TX RealTimeRunData 865.00 MHz --> 15 94 40 13 03 80 11 91 44 80 0B 00 66 23 79 55 00 00 00 00 00 00 00 00 91 9C 7A 
10:19:38.493 > RX Period End
10:19:38.493 > All missing
10:19:38.493 > Nothing received, resend whole request_

And it goes on like that. I got 2 HMS inverters and I get the same result on both. I was on firmware v23.x and upgraded to the latest, hoping it would fix the issue but nope (kinda makes sense as it worked fine on the previous firmware and then suddenly stopped out of nowhere). Only if I power cycle the opendtu it receives data from the HMS inverters but only for less than 30 min or so, then it stops receiving. The inverters still supply solar energy to the house according to my Shelly3EM. I moved the opendtu closer, increased the transmit power and such but the issue remains. As if something had changed either on the inverters (power cycled them too yesterday) or as if something in the environment had changed. Strange. I mean, the inverters cannot change their software without an official DTU, right? So technically, nothing should have changed on them either. I also turned off MQTT in opendtu for testing but the issue happens anyway.
When I receive something after power cycling the opendtu the connection to the inverters report around -50 to -60 dbm which did fine in the last few months.

I am not sure what info someone needs to troubleshoot this. I took this from opendtu:
Chip-Modell: ESP32-D0WD-V3
Chip revision: 3
Chip cores: 2
CPU frequency: 240 MHz

edit: root cause issue was a melted cable: https://github.com/tbnobody/OpenDTU/discussions/1930#discussioncomment-9589767

**Comment by Lenni** [comment]
I have the same Issue, currently I still do not have a fix for this

**Comment by SkyBird1980** [comment]
Same for me. Some Action for Input, if i disconnect Grid Power, the Inverter is reachable by OpenDTU and go into yellow state. if it returns to produce mode it will get a disconnect soon again.
Adding the Log:
[log.txt](https://github.com/tbnobody/OpenDTU/files/15286244/log.txt)
some more info, maybe it depends on the actual power. if i have over 800W the connect get lost, 
(btw: HMS-1600-4T , String 1,2,4 connected, 3 disconnected)

**Comment by Landorin-GH** [comment]
At least in my case I would narrow down the issue to the opendtu device, probably hardware related. I obtained a original DTU today and it has been connected to the inverter all day long without any issues. Could also be a software issue. Yet I had that issue with the v23.x as well as v24.x firmware and in the beginning the device ran fine for weeks.

**Comment by SkyBird1980** [comment]
The Original DTU can't be an solution for my case. I like the no Cloud feature. And this can't be a target. However it have to be opendtu related. i can see it happens if the inverter produces over ~ 800W. if you reset the hms by ac power it will be reachable again instand. then the power goes up step by step the problem will come again.

**Comment by Landorin-GH** [comment]
The original DTU was simply an idea to narrow down the issue / exclude things like some frequency distortion by another device or party.

Anyway, after quite (!) some troubleshooting the seller found the issue fortunately today:
an electrical overload caused the cables to melt which lead to undervoltage of the ESP module after a while and made it go to sleep. The only indicator outside the case was the darker than usual LED display (I wasn't aware of the difference as beginner).

I guess if the opendtu console log would / could be more verbose it might have went faster to find the cause. At the very least contain a message if the ESP module has issues, is undervolted (if it can even detect that by itself that is) or simply has gone to sleep.
